orderfilter/validate_js.go

```diff
@@ -33,13 +39,24 @@ func (s *SchemaValidationResult) Errors() []*SchemaValidationError {
 // ValidateOrderJSON Validates a JSON encoded signed order using the AJV javascript library.
 // This libarary is used to increase the performance of Mesh nodes that run in the browser.
 func (f *Filter) ValidateOrderJSON(orderJSON []byte) (*SchemaValidationResult, error) {
-	jsResult := f.orderValidator.Invoke(string(orderJSON))
-	fatal := jsResult.Get("fatal")
-	if !jsutil.IsNullOrUndefined(fatal) {
-		return nil, errors.New(fatal.String())
+	resultChan := make(chan js.Value, 1)
+	errChan := make(chan error, 1)
+	go func() {
+		time.Sleep(sleepTime)
+		result, err := jsutil.AwaitPromise(f.orderValidator.Invoke(string(orderJSON)))
```

Comment:
Can we use `AwaitPromiseContext` here? We should really be using that everywhere going forward. `AwaitPromise` will never timeout and can potentially hang forever.